v2/project/GeneratorPlugin.scala

```diff
@@ -0,0 +1,69 @@
+import sbt._
+import sbt.Keys._
+import scala.reflect.runtime.{universe => ru}
+import scala.tools.reflect.ToolBox
+
+object GeneratorPlugin extends AutoPlugin {
+  override def trigger: PluginTrigger = allRequirements
+
+  object autoImport {
+    val generateSourcesTask = taskKey[Seq[File]]("Generates source files from templates")
+  }
+  import autoImport._
```

Comment:
object `autoImport` seems to be useless.
Replace line 9 to 12 by
```val generateSourcesTask = taskKey[Seq[File]]("Generates source files from templates")```
should be fine.